### PR TITLE
Updating to dotnet 5.0

### PR DESCRIPTION
**Pull request by aasimkhan30**
Updating STS to dotnet 5.0. This will be one of the first steps to remove the several build warnings we see during our pipelines runs for using older versions of dotnet that are nearing EOL. 

General Notes: 

 1. Unfortunately, due to SRGEN not available to use with dotnet 5.0, I have added a pipeline step that installs dotnet 3.1.
 2. Will follow up with a separate PR that fixes all the code analysis warnings. 
 3. This will also require a bigger change in https://github.com/microsoft/azuredatastudio/blob/main/extensions/mssql/config.json as the archive names are getting changed.

### CODE REVIEW

File: test/Microsoft.Kusto.ServiceLayer.UnitTests/Microsoft.Kusto.ServiceLayer.UnitTests.csproj [hunk 1]
```diff
@@ -1,8 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
     <PropertyGroup>
-        <TargetFramework>netcoreapp3.1</TargetFramework>
-
+        <TargetFramework>$(TestProjectsTargetFramework)</TargetFramework>
```
Comment (aasimkhan30) [inline test/Microsoft.Kusto.ServiceLayer.UnitTests/Microsoft.Kusto.ServiceLayer.UnitTests.csproj:4]: Changing target framework of Kusto from a hardcoded value to the same as rest of the project.

File: test/Microsoft.SqlTools.ServiceLayer.UnitTests/Connection/ConnectionServiceTests.cs [hunk 2]
```diff
@@ -64,7 +64,7 @@ public void CanCancelConnectRequest()
 
             // Given a connection that times out and responds to cancellation
             var mockConnection = new Mock<DbConnection> { CallBase = true };
-            CancellationToken token;
+            CancellationToken token = new CancellationToken();
```
Comment (aasimkhan30) [inline test/Microsoft.SqlTools.ServiceLayer.UnitTests/Connection/ConnectionServiceTests.cs:67]: This was throwing a compilation error

**Comment by aasimkhan30** [comment]
Mentioning this issue https://github.com/microsoft/sqltoolsservice/issues/1217

File: azure-pipelines/build.yml [hunk 3]
```diff
@@ -43,6 +43,11 @@ steps:
     projects: '$(Build.SourcesDirectory)/src/Microsoft.Kusto.ServiceLayer'
     arguments: '--configuration Release'
 
+- task: UseDotNet@2
```
Comment (shueybubbles) [inline azure-pipelines/build.yml:46]: maybe we should rewrite srgen in go so it's not framework-dependent anymore. Volunteers? :-)

**Review by shueybubbles — APPROVED** [review]

**Comment by aasimkhan30** [comment]
@Charles-Gagnon I had tried a locally built one, but I will try one from the azure pipelines

**Comment by aasimkhan30** [comment]
@Charles-Gagnon, tried the win-x64 zip from the adhoc build. Everything seems to work fine.

**Comment by Charles-Gagnon** [comment]
Thanks, could you also test on Mac and Linux?

**Comment by udeeshagautam** [comment]
@aasimkhan30 as mentioned in chat this can affect SQL project build since it used the dotnet dlls at runtime. There can be two impacts (I am not sure if there certainly will be or not)
1) we might need users to depend on dotnet 5 for build (customer impact)
2) we might also need to update dacfx which would be a bigger issue. 

Please wait for for a validation from our side.

**Review by kisantia — APPROVED** [review]
I verified that the projects and dacpac extensions are still working with these changes.

**Review by JustinMDotNet — APPROVED** [review]
Looks good. Tested...

Connect to Kusto Cluster
Execute Query
Generate Take 10 Script
Generate Function Script
Execute Function
Connect to Notebook
Execute query in Notebook
Hover in Query
Hover in Notebook
Autocomplete in Query
Autocomplete in Notebook
Dashboard
Navigating object explorer

**Comment by aasimkhan30** [comment]
Verified the build on mac, linux and ubuntu and everything seems to work. Pushing this change in.

**Pull request merged**